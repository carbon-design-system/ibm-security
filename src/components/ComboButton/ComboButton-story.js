/**
 * @file Combo button stories.
 * @copyright IBM Security 2019 - 2021
 */

import { ArrowRight20, Filter20 } from '@carbon/icons-react';

import React from 'react';

import { action } from '@storybook/addon-actions';
import { radios } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import { patterns } from '../../../.storybook';

import { TooltipDirection } from '../IconButton/IconButton';
import { ComboButton, ComboButtonItem } from '../..';

const props = () => ({
  direction: radios(
    'Menu direction (direction)',
    {
      top: TooltipDirection.TOP,
      bottom: TooltipDirection.BOTTOM,
    },
    TooltipDirection.TOP
  ),
});

storiesOf(patterns('ComboButton'), module).add(
  'default',
  () => (
    <div style={{ width: 300 }}>
      <ComboButton {...props()}>
        <ComboButtonItem
          href="#"
          onClick={action(`onClick (Item 1)`)}
          renderIcon={ArrowRight20}
          target="_blank"
        >
          Item 1 -- becomes primary button and text will be truncated
        </ComboButtonItem>
        <ComboButtonItem
          href="#"
          onClick={action(`onClick (Item 2)`)}
          renderIcon={Filter20}
        >
          Item 2
        </ComboButtonItem>
        <ComboButtonItem
          href="#"
          renderIcon={Filter20}
          onClick={action(`onClick (Item 3)`)}
          data-overflow-menu-primary-focus
        >
          Item 3 -- will be selected when menu opens
        </ComboButtonItem>
        <ComboButtonItem
          href="#"
          renderIcon={Filter20}
          onClick={action(`onClick (Item 3)`)}
          target="_blank"
        >
          Item 4 -- will open link in new tab when clicked
        </ComboButtonItem>
      </ComboButton>
    </div>
  ),
  {
    info: {
      text: `
          The \`ComboButton\` accepts \`ComboButtonItem\` components as children.

          The first child of the \`ComboButton\` will be marked as the "primary" action, and will appear as a \`Button\` next to the \`OverflowMenu\` of additional actions.

          If there is only one child of \`ComboButton\`, then an \`OverflowMenu\` will not be rendered.
        `,
    },
  }
);
