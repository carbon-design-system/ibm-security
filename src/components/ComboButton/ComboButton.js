/**
 * @file Combo button.
 * @copyright IBM Security 2019 - 2020
 */

import { ChevronDown16, ChevronUp16 } from '@carbon/icons-react';

import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { createElement, useState } from 'react';

import { carbonPrefix, getComponentNamespace } from '../../globals/namespace';

import Button from '../Button';
import OverflowMenu from '../OverflowMenu';
import OverflowMenuItem from '../OverflowMenuItem';
import { TooltipDirection } from '../IconButton/IconButton';

import { namespace as buttonNamespace } from '../Button/Button';

export const namespace = getComponentNamespace('combo-button');

const ComboButton = ({
  children,
  className,
  direction,
  menuOffset,
  menuOffsetFlip,
  selectorPrimaryFocus,
}) => {
  const [isOpen, setIsOpen] = useState(false);

  const childrenArray = React.Children.toArray(children).filter(Boolean);

  // Save first child (e.g., primary action) to use as a `Button`:
  const buttonProps = childrenArray[0].props;
  // Need to explicitly define props, versus using `...rest`,
  // because otherwise unused `OverflowMenuItem`-related props from
  // may trigger invalid DOM warnings.
  const primaryActionWithProps = (
    <Button
      {...buttonProps}
      className={classnames(buttonProps.className, `${namespace}--primary`)}
      disabled={buttonProps.disabled}
      href={buttonProps.href}
      iconDescription={buttonProps.iconDescription}
      kind="primary"
      id={buttonProps.id}
      onClick={buttonProps.onClick}
      renderIcon={buttonProps.renderIcon}
      type="button"
    >
      <span
        className={`${carbonPrefix}text-truncate--end`}
        title={buttonProps.children}
      >
        {buttonProps.children}
      </span>
    </Button>
  );

  // Save remaining children to be displayed in the `OverflowMenu`:
  let overflowItems;
  let overflowMenuItemWithProps;
  if (childrenArray.length > 1) {
    overflowItems = childrenArray.slice(1);

    // Create `OverflowMenuItem` components:
    overflowMenuItemWithProps = overflowItems.map((item, index) => {
      // Need to explicitly define props, versus using `...rest`,
      // because otherwise unused `Button`-related props from
      // may trigger invalid DOM warnings.
      const {
        children,
        className,
        disabled,
        href,
        iconDescription,
        id,
        onClick,
        renderIcon: Icon,
        ...other
      } = item.props;

      return (
        <OverflowMenuItem
          className={classnames(className, `${namespace}-item__wrapper`)}
          disabled={disabled}
          href={href}
          itemText={
            <>
              <span
                className={`${carbonPrefix}text-truncate--end`}
                title={children}
              >
                {children}
              </span>
              {Icon && <Icon aria-label={iconDescription} />}
            </>
          }
          id={id}
          key={id || index}
          onClick={onClick}
          {...other}
        />
      );
    });
  }

  const renderOverflowMenuIcon = () =>
    createElement(isOpen ? ChevronUp16 : ChevronDown16, {
      className: `${namespace}__icon`,
    });

  return (
    <div
      className={classnames(namespace, className)}
      data-floating-menu-container
    >
      <div className={`${namespace}__group`}>
        {primaryActionWithProps}

        {overflowMenuItemWithProps !== undefined && (
          <OverflowMenu
            className={classnames(
              // Button-specific classes for styling:
              buttonNamespace,
              `${carbonPrefix}btn`,
              `${carbonPrefix}btn--primary`,

              // Button as a child of combo button:
              `${namespace}__button`,

              // Overflow menu as a child of combo button:
              `${namespace}__overflow-menu`
            )}
            direction={direction}
            menuOffset={menuOffset}
            menuOffsetFlip={menuOffsetFlip}
            menuOptionsClass={`${carbonPrefix}list-box__menu`}
            onClick={() => setIsOpen(!isOpen)}
            onClose={() => setIsOpen(false)}
            renderIcon={renderOverflowMenuIcon}
            selectorPrimaryFocus={selectorPrimaryFocus}
          >
            {overflowMenuItemWithProps}
          </OverflowMenu>
        )}
      </div>
    </div>
  );
};

ComboButton.propTypes = {
  /** @type {node} The child nodes. */
  children: PropTypes.node.isRequired,

  /** @type {string} Extra classes to add. */
  className: PropTypes.string,

  /** @type {string} Overflow menu direction. */
  direction: PropTypes.oneOf([TooltipDirection.TOP, TooltipDirection.BOTTOM]),

  /**
   * The adjustment in position applied to the floating menu.
   */
  menuOffset: PropTypes.oneOfType([
    PropTypes.shape({
      top: PropTypes.oneOf([PropTypes.number, PropTypes.string]),
      left: PropTypes.oneOf([PropTypes.number, PropTypes.string]),
    }),
    PropTypes.func,
  ]),

  /**
   * The adjustment in position applied to the floating menu.
   */
  menuOffsetFlip: PropTypes.oneOfType([
    PropTypes.shape({
      top: PropTypes.oneOf([PropTypes.number, PropTypes.string]),
      left: PropTypes.oneOf([PropTypes.number, PropTypes.string]),
    }),
    PropTypes.func,
  ]),

  /**
   * Specify a CSS selector that matches the DOM element that should
   * be focused when the OverflowMenu opens
   */
  selectorPrimaryFocus: PropTypes.string,
};

ComboButton.defaultProps = {
  className: '',
  direction: TooltipDirection.TOP,
  menuOffset: () => ({
    left: 'auto',
  }),
  menuOffsetFlip: undefined,
  selectorPrimaryFocus: '[data-overflow-menu-primary-focus]',
};

export default ComboButton;
