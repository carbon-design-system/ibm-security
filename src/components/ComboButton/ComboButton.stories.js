/**
 * @file Combo button stories.
 * @copyright IBM Security 2019
 */

import React from 'react';

import { storiesOf } from '@storybook/react';
import { radios } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import ArrowRight20 from '@carbon/icons-react/lib/arrow--right/20';
import Filter20 from '@carbon/icons-react/lib/filter/20';

import { patterns } from '../../../.storybook';

import { TooltipDirection } from '../IconButton/IconButton';
import { ComboButton, ComboButtonItem } from '../..';

const props = () => ({
  direction: radios(
    'Menu direction (direction)',
    {
      top: TooltipDirection.TOP,
      bottom: TooltipDirection.BOTTOM,
    },
    TooltipDirection.TOP
  ),
});

storiesOf(patterns('ComboButton'), module).add(
  'default',
  () => (
    <div style={{ width: 300 }}>
      <ComboButton {...props()}>
        <ComboButtonItem href="#" renderIcon={ArrowRight20}>
          Item 1 (becomes primary button and text will be truncated)
        </ComboButtonItem>
        {Array(5)
          .fill(0)
          .map((item, index) => {
            const text = `Item ${index +
              2} - text may be long and will be truncated`;
            return (
              <ComboButtonItem
                className="some-class"
                key={text}
                href={`#${index}`}
                index={index}
                id={index}
                onClick={action(`onClick (${text})`)}
                renderIcon={index % 2 === 0 ? Filter20 : null} // Show icon at even indexes.
              >
                {text}
              </ComboButtonItem>
            );
          })}
      </ComboButton>
    </div>
  ),
  {
    info: {
      text: `
          The \`ComboButton\` accepts \`ComboButtonItem\` components as children.

          The first child of the \`ComboButton\` will be marked as the "primary" action, and will appear as a \`Button\` next to the \`OverflowMenu\` of additional actions.

          If there is only one child of \`ComboButton\`, then an \`OverflowMenu\` will not be rendered.
        `,
    },
  }
);
