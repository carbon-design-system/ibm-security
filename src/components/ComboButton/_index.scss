////
/// Combo button component.
/// @group combo-button
/// @copyright IBM Security 2019
////

@import '@carbon/layout/scss/mini-unit';

@import '@carbon/layout/scss/spacing';
@import '@carbon/themes/scss/tokens';

@import 'carbon-components/scss/globals/scss/helper-mixins';

@import '../Button/index';
@import '../Dropdown/index';
@import '../OverflowMenu/index';

@import '../Component/mixins';

@include security--component($name: combo-button) {
  $root: &;

  display: inline-flex;
  position: relative;

  & &--primary {
    @include text-overflow();
    display: flex;
    flex: 1;
  }

  &__group {
    display: flex;
    justify-content: flex-end;
    width: 100%;
  }

  &__icon {
    fill: $icon-03;
  }

  & &__overflow-menu {
    width: 3rem;
    background-color: $interactive-01;
    margin-left: 1px;

    &:hover,
    &.#{$prefix}--overflow-menu--open {
      background-color: $hover-primary;
    }

    &:active {
      background-color: $active-primary;
    }
  }

  .#{$prefix}--overflow-menu-options {
    display: grid;
    width: 100%;
    background-color: $ui-02;
    outline: unset;

    &__option {
      display: inherit;

      &:hover {
        background-color: $hover-selected-ui;
      }

      &:first-child #{$root}-item__wrapper {
        border-top-color: transparent;
      }
    }

    &::after {
      display: none;
    }
  }

  &-item {
    width: 100%;
    display: flex;
    justify-content: space-between;
  }

  & &-item__wrapper {
    @include text-overflow();
    outline: unset;
    padding: 0;
    margin: 0 1rem;
    border-top: 1px solid $active-ui;
    border-bottom: 1px solid transparent;
    display: flex;
    width: unset;
    max-width: unset;
    justify-content: space-between;

    &:focus {
      @include focus-outline('outline');
      margin: 0;
      padding: 0 1rem;
      border-color: transparent;
    }

    &:hover {
      color: $text-01;
      background-color: unset;
    }

    & > span + svg {
      margin-left: $carbon--spacing-03;
      flex-shrink: 0;
    }

    &:disabled,
    &:disabled:hover {
      color: $text-03;

      svg {
        fill: $text-03;
      }
    }
  }
}
