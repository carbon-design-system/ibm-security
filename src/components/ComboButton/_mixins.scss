////
/// Combo button component.
/// @group combo-button
/// @copyright IBM Security 2019
////

@import '../../globals/index';

@mixin combo-button {
  $root: &;

  position: relative;
  display: flex;
  justify-content: flex-end;
  width: 100%;

  & &--primary {
    @include text-overflow();
    display: flex;
    flex: 1;
  }

  &__icon {
    fill: $icon-03;
  }

  & &__overflow-menu {
    width: 3rem;
    margin-left: 1px;

    &:hover,
    &.#{$prefix}--overflow-menu--open {
      background-color: $hover-primary;
    }

    &:active {
      background-color: $active-primary;
    }
  }

  .#{$prefix}--overflow-menu-options {
    display: grid;
    width: 100%;
    background-color: $ui-02;
    outline: unset;

    &__option {
      display: inherit;

      &:hover {
        background-color: $hover-selected-ui;
      }

      &:first-child #{$root}-item__wrapper {
        border-top-color: transparent;
      }
    }

    &::after {
      display: none;
    }
  }

  &-item {
    width: 100%;
    display: flex;
    justify-content: space-between;
  }

  & &-item__wrapper {
    @include text-overflow();
    outline: unset;
    padding: 0;
    margin: 0 1rem;
    border-top: 1px solid $active-ui;
    border-bottom: 1px solid transparent;
    display: flex;
    width: unset;
    max-width: unset;
    justify-content: space-between;

    &:focus {
      @include focus-outline('outline');
      margin: 0;
      padding: 0 1rem;
      border-color: transparent;
    }

    &:hover {
      color: $text-01;
      background-color: unset;
    }

    & > span + svg {
      margin-left: $carbon--spacing-03;
      flex-shrink: 0;
    }

    &:disabled,
    &:disabled:hover {
      color: $text-03;

      svg {
        fill: $text-03;
      }
    }
  }
}
