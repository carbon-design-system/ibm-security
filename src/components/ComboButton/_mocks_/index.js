/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
