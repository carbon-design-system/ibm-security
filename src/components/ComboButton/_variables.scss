////
/// Combo button variables.
/// @group combo-button
/// @copyright IBM Security 2020
////

/// Name.
/// @type String
/// @access private
$combo-button__name: combo-button;
