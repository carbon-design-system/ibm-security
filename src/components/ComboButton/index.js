/**
 * @file Combo button entry point.
 * @copyright IBM Security 2019
 */

export default from './ComboButton';
export ComboButtonItem from './ComboButtonItem';

// export { OverflowMenuItem as ComboButtonItem } from '../OverflowMenuItem';
