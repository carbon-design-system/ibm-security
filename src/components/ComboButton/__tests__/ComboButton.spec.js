/**
 * @file Combo button tests.
 * @copyright IBM Security 2019
 */
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import ArrowRight20 from '@carbon/icons-react/lib/arrow--right/20';

import ComboButton, { ComboButtonItem } from '../';

const COMBO_BUTTON_TESTID = 'combo-button';

const renderComboButton = (overflowMenuItemCount = 0) => {
  const primaryItem = (
    <ComboButtonItem
      key="test-primary-item"
      id="test-primary-item"
      renderIcon={ArrowRight20}
    >
      Primary task
    </ComboButtonItem>
  );
  const overflowMenuItems = new Array(overflowMenuItemCount)
    .fill(undefined)
    .map((_, index) => {
      const key = `test-menu-item-${index}`;
      return (
        <ComboButtonItem key={key} id={key}>
          Other task
        </ComboButtonItem>
      );
    });

  const items = [primaryItem, ...overflowMenuItems];

  return render(
    <div data-testid={COMBO_BUTTON_TESTID}>
      <ComboButton>{items}</ComboButton>
    </div>
  );
};

const getComboBox = () => screen.getByTestId(COMBO_BUTTON_TESTID);
const getOverflowMenuButton = () => screen.getByLabelText('Menu');
const clickComboBox = overflowMenuButton => {
  userEvent.click(overflowMenuButton);
};

describe('ComboButton', () => {
  it('renders a combo button without an overflow menu', () => {
    renderComboButton(0);
    const comboButton = getComboBox();

    expect(comboButton).toMatchSnapshot();
  });

  it('renders a combo button with children and an overflow menu', () => {
    renderComboButton(2);
    const comboButton = getComboBox();

    expect(comboButton).toMatchSnapshot();
  });

  it('renders overflow menu items when overflow menu is clicked (opened) and removes overflow menu items when clicked again (closed)', () => {
    const MENU_ITEM_COUNT = 2;
    const getMenuItemsLength = () => screen.queryAllByRole('menuitem').length;

    renderComboButton(MENU_ITEM_COUNT);
    const overflowMenuButton = getOverflowMenuButton();

    /*
      Let's test the opening and closing of the ComboButton menu 
    */
    expect(getMenuItemsLength()).toEqual(0); // Menu items should not be shown
    clickComboBox(overflowMenuButton); // Open the menu
    expect(getMenuItemsLength()).toEqual(MENU_ITEM_COUNT); // Menu items should now be shown
    clickComboBox(overflowMenuButton); // Close the Menu
    expect(getMenuItemsLength()).toEqual(0); // Menu items should not be shown again
  });

  it('should have no Axe or DAP violations', async () => {
    renderComboButton(4);

    const comboButton = getComboBox();
    const assertHasNoViolations = async () => {
      await expect(comboButton).toHaveNoAxeViolations();
      await expect(comboButton).toHaveNoDAPViolations('ComboButton');
    };

    await assertHasNoViolations();

    // Now let's open the ComboButton menu and re-assert conditions
    const overflowMenuButton = getOverflowMenuButton();
    clickComboBox(overflowMenuButton);

    await assertHasNoViolations();
  });
});
