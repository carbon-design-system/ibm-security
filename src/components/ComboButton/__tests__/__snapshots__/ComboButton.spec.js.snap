// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboButton renders a combo button with children and an overflow menu 1`] = `
<ComboButton
  className=""
  direction="top"
>
  <div
    className="security--combo-button"
    data-floating-menu-container={true}
  >
    <Button
      className="security--combo-button--primary"
      disabled={false}
      iconDescription=""
      kind="primary"
      largeText={null}
      loading={false}
      onClick={[Function]}
      renderIcon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      tabIndex={0}
      type="button"
    >
      <ForwardRef(Button)
        className="security--button security--combo-button--primary"
        disabled={false}
        iconDescription=""
        kind="primary"
        onClick={[Function]}
        renderIcon={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        tabIndex={0}
        type="button"
      >
        <button
          className="security--button security--combo-button--primary bx--btn bx--btn--primary"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bx--text-truncate--end"
            title="Start a task"
          >
            Start a task
          </span>
          <ForwardRef(ArrowRight20)
            aria-hidden="true"
            aria-label=""
            className="bx--btn__icon"
          >
            <Icon
              aria-hidden="true"
              aria-label=""
              className="bx--btn__icon"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 20 20"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <svg
                aria-hidden={true}
                aria-label=""
                className="bx--btn__icon"
                focusable="false"
                height={20}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
                viewBox="0 0 20 20"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.8 2.8l-1 1 5.4 5.5H1v1.4h15.2l-5.4 5.5 1 1L19 10z"
                />
              </svg>
            </Icon>
          </ForwardRef(ArrowRight20)>
        </button>
      </ForwardRef(Button)>
    </Button>
    <ForwardRef(OverflowMenu)
      className="security--button bx--btn bx--btn--primary security--combo-button__button security--combo-button__overflow-menu"
      direction="top"
      menuOffset={[Function]}
      menuOptionsClass="bx--list-box__menu"
      onClose={[Function]}
      onOpen={[Function]}
      renderIcon={[Function]}
    >
      <OverflowMenu
        ariaLabel="Menu"
        className="security--button bx--btn bx--btn--primary security--combo-button__button security--combo-button__overflow-menu"
        direction="top"
        flipped={false}
        iconDescription="open and close list of options"
        innerRef={null}
        menuOffset={[Function]}
        menuOffsetFlip={[Function]}
        menuOptionsClass="bx--list-box__menu"
        onClick={[Function]}
        onClose={[Function]}
        onKeyDown={[Function]}
        onOpen={[Function]}
        open={false}
        renderIcon={[Function]}
        tabIndex={0}
      >
        <ClickListener
          onClickOutside={[Function]}
        >
          <div
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="Menu"
            className="security--button bx--btn bx--btn--primary security--combo-button__button security--combo-button__overflow-menu bx--overflow-menu"
            onClick={[Function]}
            onClose={[Function]}
            onKeyDown={[Function]}
            open={false}
            role="button"
            tabIndex={0}
          >
            <renderIcon
              aria-label="open and close list of options"
              className="bx--overflow-menu__icon"
              focusable="false"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              <ForwardRef(ChevronDown16)
                className="security--combo-button__icon"
              >
                <Icon
                  className="security--combo-button__icon"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    aria-hidden={true}
                    className="security--combo-button__icon"
                    focusable="false"
                    height={16}
                    preserveAspectRatio="xMidYMid meet"
                    style={
                      Object {
                        "willChange": "transform",
                      }
                    }
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                    />
                  </svg>
                </Icon>
              </ForwardRef(ChevronDown16)>
            </renderIcon>
          </div>
        </ClickListener>
      </OverflowMenu>
    </ForwardRef(OverflowMenu)>
  </div>
</ComboButton>
`;

exports[`ComboButton renders a combo button without an overflow menu 1`] = `
<ComboButton
  className=""
  direction="top"
>
  <div
    className="security--combo-button"
    data-floating-menu-container={true}
  >
    <Button
      className="security--combo-button--primary"
      disabled={false}
      iconDescription=""
      kind="primary"
      largeText={null}
      loading={false}
      onClick={[Function]}
      renderIcon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      tabIndex={0}
      type="button"
    >
      <ForwardRef(Button)
        className="security--button security--combo-button--primary"
        disabled={false}
        iconDescription=""
        kind="primary"
        onClick={[Function]}
        renderIcon={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        tabIndex={0}
        type="button"
      >
        <button
          className="security--button security--combo-button--primary bx--btn bx--btn--primary"
          disabled={false}
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bx--text-truncate--end"
            title="Start a task"
          >
            Start a task
          </span>
          <ForwardRef(ArrowRight20)
            aria-hidden="true"
            aria-label=""
            className="bx--btn__icon"
          >
            <Icon
              aria-hidden="true"
              aria-label=""
              className="bx--btn__icon"
              height={20}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 20 20"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <svg
                aria-hidden={true}
                aria-label=""
                className="bx--btn__icon"
                focusable="false"
                height={20}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "willChange": "transform",
                  }
                }
                viewBox="0 0 20 20"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.8 2.8l-1 1 5.4 5.5H1v1.4h15.2l-5.4 5.5 1 1L19 10z"
                />
              </svg>
            </Icon>
          </ForwardRef(ArrowRight20)>
        </button>
      </ForwardRef(Button)>
    </Button>
  </div>
</ComboButton>
`;
